lib/test.js

```diff
@@ -58,7 +59,8 @@ Test.prototype.serverAddress = function(app, path){
   if (!addr) this._server = app.listen(0);
   var port = app.address().port;
   var protocol = app instanceof https.Server ? 'https' : 'http';
-  return protocol + '://127.0.0.1:' + port + path;
+  var hostname = this._options.isLocalhost ? 'localhost': '127.0.0.1';
```

Comment:
How about this?
var hostname = !!this._options.hostname? this._options.hostname: '127.0.0.1';